tests/kokkos-based/helpers.hpp

```diff
@@ -166,60 +195,216 @@ bool is_same_vector(
   return is_same_vector(make_mdspan(v1), make_mdspan(v2));
 }
 
-template <typename RealValue>
-KOKKOS_INLINE_FUNCTION
-RealValue scalar_diff(RealValue v1, RealValue v2) {
-  return std::abs(v2 - v1);
+template <typename ElementType1,
+          typename LayoutPolicy1,
+          typename AccessorPolicy1,
+          typename ElementType2,
+          typename LayoutPolicy2,
+          typename AccessorPolicy2>
+auto vector_abs_diff(
+    mdspan<ElementType1, extents<dynamic_extent>, LayoutPolicy1, AccessorPolicy1> v1,
+    mdspan<ElementType2, extents<dynamic_extent>, LayoutPolicy2, AccessorPolicy2> v2)
+{
+  using RetType = decltype(scalar_abs_diff(v1[0], v2[0])); // will be same for views
+  const auto size = v1.extent(0);
+  if (size != v2.extent(0))
+    return std::numeric_limits<RetType>::max(); // very, very different
+  const auto v1_view = KokkosKernelsSTD::Impl::mdspan_to_view(v1);
+  const auto v2_view = KokkosKernelsSTD::Impl::mdspan_to_view(v2);
+  RetType diff = static_cast<RetType>(0);
+  Kokkos::parallel_reduce(size,
+    KOKKOS_LAMBDA(const std::size_t i, RetType &diff){
```

Comment:
Would you consider giving the inner `diff` variable a different name, to prevent shadowing warnings and make the code more clear?

Reply:
> Would you mind if I use the reducer's `join()` directly ?

I would prefer that, as a way to avoid code duplication -- thanks!